firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -376,7 +392,7 @@ public SnapshotVersion getLastRemoteSnapshotVersion() {
             queryCache.setLastRemoteSnapshotVersion(remoteVersion);
           }
 
-          return localDocuments.getDocuments(changedDocKeys);
+          return localDocuments.getDocuments(changedDocs);
```

Comment:
Optimization 3 ("No double get"): avoid retrieving the documents from local database again, we already have them in this function; just apply the pending write batches to them.